### PR TITLE
Update template/example

### LINKED ISSUE #7: Document configuration items
We should formally document the configuration items for each plugin.

### PR DESCRIPTION
**Pull request by astrofrog**
The current example was out of date, so I decided to just make it a bit more generic so that it can be copied/pasted and edited, and renamed the directory to ``template``.

Fixes #7

### CODE REVIEW

**Comment by astrofrog** [comment]
I've now also added some docs about the plugins and configuration items

**Comment by astrofrog** [comment]
Ok so I was like 'why not just finish writing all the docs' and here we go.

File: doc/plugins.rst [hunk 1]
```diff
@@ -1,2 +1,147 @@
...
+<https://circleci.com/docs/2.0/artifacts/>`_. The baldrick plugin will
+automatically post the link to the artifacts as a status check in a GitHub pull
+request to avoid having to click through multiple pages to find the link to the
+artifacts. To enable this plugin, include the following in your
+``pyproject.toml`` file::
+
+    [ tool.<your-bot-name>.circleci_artifacts ]
+    enabled = true
+
+There are no further configuration options for this plugin.
```
Comment (Cadair) [inline doc/plugins.rst]: There are extra config options! You can specify what artifacts it reports on. It defaults to `docs/build/index.html` but you can customise it i.e. https://github.com/matrix-org/matrix-doc/blob/master/pyproject.toml

Comment (astrofrog) [reply doc/plugins.rst]: Fixed

File: doc/plugins.rst [hunk 2]
```diff
@@ -1,2 +1,147 @@
...
+* ``all_passed_message = "..."``: the message to show in a comment if all checks passed.
+
+* ``pull_request_substring``: a string that can be used to identify previous
+  comments posted by the bot. This should be a string common to
+  ``all_passed_message``, and ``fail_prologue`` or ``fail_epilogue``.
+
+GitHub milestone checker
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+One of the pull request handler plugins checks whether the milestone has been
```
[the remark this answers is not in the document]
Comment (astrofrog) [reply doc/plugins.rst]: Fixed

File: doc/plugins.rst [hunk 3]
```diff
@@ -1,2 +1,147 @@
...
+    enabled = true
+
+If you wish to customize the message shown in the results of the check, you can
+use the ``missing_message = "..."`` and ``present_message = "..."`` configuration
+items.
+
+Towncrier changelog checker
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Another built-in pull request handler plugin can be used to check that towncrier
```
[the remark this answers is not in the document]
Comment (astrofrog) [reply doc/plugins.rst]: Added

**Pull request merged**